### PR TITLE
Update

### PR DESCRIPTION
**Pull request by y0no**
Add Obfuscation mode
Add Mustache face (obfuscated characters)

### CODE REVIEW

File: emoticon.coffee [hunk 1]
```diff
@@ -45,14 +52,22 @@ class Parser
         match = match[1]
         token = new Instruction match, 'emoticon'
         source.push token
-      else if match = code.match(rNumber)
+      else if match = code.match rMustache
```
Comment (amasad) [inline emoticon.coffee:55]: Shouldn't we check if we are in obfuscated mode before we match the Mustache face?

File: emoticon.coffee [hunk 2]
```diff
@@ -171,6 +186,11 @@ class Interpreter
       @putRight @currentList(), instruction.value
       @lists['X'][0]++
       
+    else if instruction.type == 'mustache'
+      @putRight  @currentList(), @parseMustache instruction.value
+      console.log(this.lists)
```
Comment (amasad) [inline emoticon.coffee]: probably forgot to remove this

**Comment by amasad** [comment]
Awesome. Love it. :D I left a couple of comments inline.

File: emoticon.coffee (hunk 1 above)
Comment (y0no) [inline emoticon.coffee:55]: It seems that obfuscated mode and obfuscated char are different. Obfuscated mode supresse all characters that are not recognized. Obfuscated characters are just a way to hide strings. I don't know if it necessary to be on obfuscated mode to put obfuscated characters.

**Comment by y0no** [comment]
I've take in consideration your comments to correct the code :)

File: emoticon.coffee (hunk 1 above)
Comment (amasad) [inline emoticon.coffee:55]: ok that makes sense!

**Comment by amasad** [comment]
Thanks!
Feel free to add your name to the readme.
We should maybe add couple of test cases for the obfuscated mode and chars in the `tests/tests.coffee`.

**Comment by y0no** [comment]
Ok no problem I will add test cases as soon as possible.

**Pull request merged**